Title: Don't Preserve Bolding Under Cursor
Author: Eric Pruitt (https://www.codevat.com/, https://github.com/ericpruitt/)

When a character is emboldened, its color may not be the same as the non-bold
version, so swapping the background color and foreground color will result in
text's color becoming something other than the background color. This patch
disables preservation of the bold attribute of the character under the cursor
to resolve this.

diff --git x.c x.c
index d43a529..613c61c 100644
--- x.c
+++ x.c
@@ -1410,7 +1410,7 @@ xdrawcursor(int cx, int cy, Glyph g, int ox, int oy, Glyph og)
 	/*
 	 * Select the right color for the right mode.
 	 */
-	g.mode &= ATTR_BOLD|ATTR_ITALIC|ATTR_UNDERLINE|ATTR_STRUCK|ATTR_WIDE;
+	g.mode &= ATTR_ITALIC|ATTR_UNDERLINE|ATTR_STRUCK|ATTR_WIDE;
 
 	if (IS_SET(MODE_REVERSE)) {
 		g.mode |= ATTR_REVERSE;
